src/lib/collectors/shared/launcher.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @fileoverview Launches the given browser with the right configuration to be used via the Chrome Debugging Protocol
+ *
+ * Supported browsers: Chrome
+ *
+ */
+
+/* eslint-disable no-sync */
+
+import * as fs from 'fs';
+import * as net from 'net';
+import * as path from 'path';
+
+import * as lockfile from 'lockfile';
+import { promisify } from 'util';
+
+import { BrowserInfo, ILauncher, LauncherOptions } from '../../types';
+import { debug as d } from '../../utils/debug';
+
+const lock = promisify(lockfile.lock);
+const unlock = promisify(lockfile.unlock);
+const debug: debug.IDebugger = d(__filename);
+
+// ------------------------------------------------------------------------------
+// Common
+// ------------------------------------------------------------------------------
+export abstract class Launcher implements ILauncher {
+    private pidFile: string = path.join(process.cwd(), 'cdp.pid');
+    protected port: number = 9222;
+    private retryDelay: number = 500;
+    protected options: LauncherOptions = null;
+
+    constructor(options: LauncherOptions) {
+        this.options = options;
+    }
+
+    /** If a browser is already running, it returns its pid. Otherwise return value is -1.  */
+    private getBrowserInfo(): BrowserInfo {
+        let result = {
+            pid: -1,
+            port: this.port
+        };
+
+        try {
+            result = JSON.parse(fs.readFileSync(this.pidFile, 'utf8'));
```

Comment:
The calling function is asynchronous. I think you should `promisify` the `fs.readFile` and use that instead.